netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -423,7 +839,479 @@ net_ebpf_extension_sock_addr_authorize_connection_classify(
+    ASSERT((compartment_id == UNSPECIFIED_COMPARTMENT_ID) || (compartment_id == sock_addr_ctx->compartment_id));
+    if (compartment_id != UNSPECIFIED_COMPARTMENT_ID && compartment_id != sock_addr_ctx->compartment_id) {
+        // The client is not interested in this compartment Id. Change action to PERMIT.
+        NET_EBPF_EXT_LOG_MESSAGE_UINT32(
+            NET_EBPF_EXT_TRACELOG_LEVEL_VERBOSE,
+            NET_EBPF_EXT_TRACELOG_KEYWORD_SOCK_ADDR,
+            "The cgroup_sock_addr eBPF program is not interested in this compartment ID",
+            sock_addr_ctx->compartment_id);
+
+        action = FWP_ACTION_PERMIT;
+        goto Exit;
+    }
+
+    _net_ebpf_ext_connection_context_initialize_key(
+        &connection_context_original->key,
+        sock_addr_ctx,
+        incoming_metadata_values->transportEndpointHandle,
+        true,
+        v4_mapped);
+
+    if (net_ebpf_extension_hook_invoke_program(attached_client, sock_addr_ctx, &verdict) != EBPF_SUCCESS) {
+        status = STATUS_UNSUCCESSFUL;
+        goto Exit;
+    }
+
+    // Initialize connection_context_redirected destination with the redirected address.
+    _net_ebpf_ext_connection_context_initialize_key(
+        &connection_context_redirected->key,
+        sock_addr_ctx,
+        incoming_metadata_values->transportEndpointHandle,
+        false,
+        v4_mapped);
+
+    if (v4_mapped) {
+        sock_addr_ctx->family = AF_INET6;
+        IN_ADDR v4_address = *((IN_ADDR*)&sock_addr_ctx->user_ip4);
+        IN6_SET_ADDR_V4MAPPED((IN6_ADDR*)&sock_addr_ctx->user_ip6, (IN_ADDR*)&v4_address);
+    }
+
+    status = _net_ebpf_ext_process_redirect_verdict(
+        verdict,
+        sock_addr_ctx_original,
+        sock_addr_ctx,
+        filter,
+        classify_handle,
+        redirect_handle,
+        &redirected,
+        classify_output);
+    if (!NT_SUCCESS(status)) {
+        goto Exit;
+    }
+
+CreateContext:
+    if (verdict == BPF_SOCK_ADDR_VERDICT_PROCEED) {
+        _net_ebpf_ext_connection_context_initialize_value(redirected, verdict, connection_context_redirected);
+
+        _net_ebpf_ext_insert_connection_context_to_list(connection_context_redirected);
+
+        if (redirected) {
+            // If the connection has been redirected, then initialize connection context
+            // with original destination also.
+            _net_ebpf_ext_connection_context_initialize_value(redirected, verdict, connection_context_original);
+
+            _net_ebpf_ext_insert_connection_context_to_list(connection_context_original);
+        } else {
+            ExFreePool(connection_context_original);
+            connection_context_original = NULL;
+        }
+    } else {
+        ExFreePool(connection_context_original);
+        connection_context_original = NULL;
+        ExFreePool(connection_context_redirected);
+        connection_context_redirected = NULL;
+    }
+
+    NET_EBPF_EXT_LOG_SOCK_ADDR_CLASSIFY(
+        NET_EBPF_EXT_TRACELOG_LEVEL_INFO,
```

Comment:
verbose instead of info?

Reply:
Shouldn't the verdict for each connection be at info level?